server/api/utils/backgroundTaskWorker.py

```diff
@@ -0,0 +1,53 @@
+import asyncio
+
+
+class BackgroundTaskWorker:
+
+    """docs:
+        - https://docs.python.org/3/library/asyncio-eventloop.html#
+        - https://pymotw.com/3/asyncio/
+        - https://medium.freecodecamp.org/a-guide-to-asynchronous-
+          programming-in-python-with-asyncio-232e2afa44f6
+        - https://youtu.be/btCvFt_JsEo
+        - https://medium.com/@gauravbaliyan/a-good-use-of-asyncio-
+          in-python-and-django-8aa7bc401e5f
+        - https://snarky.ca/how-the-heck-does-async-await-work-in-python-3-5/
+        - https://blenderartists.org/t/running-background-jobs-
+          with-asyncio/673805
+        - https://stackoverflow.com/questions/46727787/
+          runtimeerror-there-is-no-current-event-loop-in-thread-in-async-apscheduler
+    """
+
+    @classmethod
+    def start_work(cls, asyn_function, args):
+
+        """This function helps to run an unpredictable function in the background.
+
+        it uses an event loop as central executor.
+
+        :param asyn_function: An asyn function that acts as coroutine.
+        :param args: The async function args or params.
+                     it should be a tuple of args.
+        :return: None
+        """
+        event_loop = asyncio.new_event_loop()
+        asyncio.set_event_loop(event_loop)
+        event_loop.run_in_executor(None, cls.run_task, asyn_function, args)
```

Comment:
Just to be sure that you fully know the reason for using the default executor here and why? 

 The `ThreadPoolExecuto`r and the `ProcessPoolExecutor`
Default: ThreadPoolExecutor

Reply:
Thanks, @andela-sjames  By `default`, an event loop uses a `ThreadPoolExecutor` which is abstracted by `asyncio`. This default executor helps to execute calls asynchronously in a pool of threads. Executing calls asynchronous in a thread is much better because it doesn't use a separate memory. I hope have been able to give the right response to the question.

Reply:
👍 
In Addition to that,  the `ThreadPoolExecutor` uses the pool of threads to execute tasks assigned to it and is generally better at `CPU-bound operations` rather than I/O bound operations.

Reply:
@tosmak16 ☝️